nomad/acl_endpoint_test.go

```diff
@@ -2145,6 +2145,128 @@ func TestACL_ListRoles(t *testing.T) {
 	err = msgpackrpc.CallWithCodec(codec, structs.ACLListRolesRPCMethod, aclRoleReq4, &aclRoleResp4)
 	require.NoError(t, err)
 	require.Len(t, aclRoleResp4.ACLRoles, 2)
+
+	// Now test a blocking query, where we wait for an update to the list which
+	// is triggered by a deletion.
+	resultCh := make(chan *structs.ACLRolesListResponse)
+
+	go func(resultCh chan *structs.ACLRolesListResponse) {
+
+		aclRoleReq5 := &structs.ACLRolesListRequest{
+			QueryOptions: structs.QueryOptions{
+				Region:        DefaultRegion,
+				AuthToken:     aclRootToken.SecretID,
+				MinQueryIndex: aclRoleResp4.Index,
+				MaxQueryTime:  10 * time.Second,
+			},
+		}
+		var aclRoleResp5 structs.ACLRolesListResponse
+		err = msgpackrpc.CallWithCodec(codec, structs.ACLListRolesRPCMethod, aclRoleReq5, &aclRoleResp5)
+		require.NoError(t, err)
+		resultCh <- &aclRoleResp5
```

Comment:
If we fail `require.NoError` here we won't ever send on the `resultCh` and I'm pretty sure that'll block the main test goroutine? We might want to either use `assert` here (and then make sure the main goroutine can handle a nil `resultCh`) or send the error back as well and have the main goroutine `select` over either an error channel or the `resultCh`.

(Same for `TestACL_GetRolesByID` below.)

Reply:
This change can be seen here: https://github.com/hashicorp/nomad/blob/f-gh-13120-acl-role-replication/nomad/acl_endpoint_test.go#L2151-L2179

Reply:
Nice!